pulp_smash/tests/rpm/api_v2/test_schedule_consumer.py

```diff
@@ -0,0 +1,163 @@
+# coding=utf-8
+"""Test that verify scheduled RPM installation and un-installation."""
+import datetime
+import random
+import time
+import unittest
+from urllib.parse import urljoin
+
+from pulp_smash import api, cli, config, exceptions, selectors, utils
+from pulp_smash.constants import (
+    CONSUMERS_PATH,
+    REPOSITORY_PATH,
+    RPM_UNSIGNED_FEED_URL,
+)
+from pulp_smash.tests.rpm.api_v2.utils import gen_distributor, gen_repo
+from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+REQUIRED_SERVICES = ('goferd',)
+
+REQUIRED_PACKAGES = (
+    'pulp-agent',
+    'pulp-consumer-client',
+    'pulp-rpm-consumer-extensions',
+    'pulp-rpm-handlers',
+    'pulp-rpm-yumplugins',
+    'python-gofer-qpid',
+)
+
+
+class ScheduledTaskConsumer(unittest.TestCase):
+    """Verify scheduled RPM installation and un-installation work.
+
+    Test the installation and un-installation of scheduled package in a
+    consumer.
+    This test case targets:
+    * `Pulp #2680 <https://pulp.plan.io/issues/2680>`_
+    * `Pulp Smash #611 <https://github.com/PulpQE/pulp-smash/issues/611>`_
+    """
+
+    def setUp(self):
+        """Install required packages and start the ``goferd`` service.
+
+        This is done because ``goferd`` must be running in order to communicate
+        with ``pulp-consumer-client``.
+        """
+        self.cfg = config.get_config()
+        if selectors.bug_is_untestable(2680, self.cfg.version):
+            self.skipTest('https://pulp.plan.io/issues/2680')
+        self.pkg_not_installed = []
+        cli_client = cli.Client(self.cfg)
+        for package in REQUIRED_PACKAGES:
+            try:
+                cli_client.run(('rpm', '-q', package))
+            except exceptions.CalledProcessError:
+                self.pkg_not_installed.append(package)
+        self.pkg_mgr = cli.PackageManager(self.cfg)
+        if self.pkg_not_installed:
+            self.pkg_mgr.install(self.pkg_not_installed)
+        system = self.cfg.get_systems('shell')[0]
+        self.svc_mgr = cli.ServiceManager(self.cfg, pulp_system=system)
+        self.svc_mgr.start(REQUIRED_SERVICES)
+        self.sudo = () if utils.is_root(self.cfg) else ('sudo',)
+
+    def tearDown(self):
+        """Remove registered consumer and stop ``goferd`` service.
+
+        Un-install previous installed packages.
+        """
+        self.svc_mgr.stop(REQUIRED_SERVICES)
+        cli.Client(self.cfg).run(self.sudo + (
+            'pulp-consumer', '-u', self.cfg.pulp_auth[0], '-p',
+            self.cfg.pulp_auth[1], 'unregister', '--force'
+        ))
+        if self.pkg_not_installed:
+            self.pkg_mgr.uninstall(self.pkg_not_installed)
```

Comment:
`pkg_not_installed` is a list of packages, right? If so, then this function invocation should use the `*` operator, so that the list will be unpacked into a series of positional arguments:

```python
self.pkg_mgr.uninstall(*self.pkg_not_installed)
```